src/main/java/org/candlepin/service/exception/subscription/SubscriptionSuccessfulActivationException.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2009 - 2023 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.service.exception.subscription;
+
+
+/**
+ * The SubscriptionSuccessfulActivationException is used to show the success of an activation attempt
+ * for a subscription.
+ */
+public class SubscriptionSuccessfulActivationException extends SubscriptionServiceException {
```

Comment:
I did a double-take here and asked "Wait, do the Adapters actually do this?" And sure enough, yes it does.

In fact, the `activateSubscription` method never returns, it always throws an exception. In this case, a successful exception. Which seems wacky to me.

We'll have more opportunities to improve this adapter later, I think we can keep all the exceptions. What are your thoughts on putting all these subscription activation exceptions in a sub package, say `org.candlepin.exception.subscription.activation` to keep them separate from the more traditional subscription exceptions?

(Another aside, what are thoughts on extracting the activation methods and exceptions out of the `SubscriptionServiceAdapter` and into a `SubscriptionActivationServiceAdapter`? This would be effort for a different, future ticket, not this one.)

Reply:
Updated both ends such that there is no exception returned on a successful action.